glib/src/boxed_any_object.rs

```diff
@@ -0,0 +1,178 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+use crate as glib;
+use crate::subclass::prelude::*;
+use crate::{object_subclass, wrapper, Object};
+use std::any::Any;
+use std::cell::{Ref, RefCell, RefMut};
+
+#[derive(thiserror::Error, Debug)]
+pub enum BorrowError {
+    #[error("type of the inner value is not as requested")]
+    InvalidType,
+    #[error("value is already mutably borrowed")]
+    AlreadyBorrowed(#[from] std::cell::BorrowError),
+}
+#[derive(thiserror::Error, Debug)]
+pub enum BorrowMutError {
+    #[error("type of the inner value is not as requested")]
+    InvalidType,
+    #[error("value is already immutably borrowed")]
+    AlreadyMutBorrowed(#[from] std::cell::BorrowMutError),
+}
+
+mod imp {
+    use super::*;
+
+    #[derive(Debug)]
+    pub struct BoxedAnyObject {
+        pub value: RefCell<Box<dyn Any>>,
+    }
+    #[object_subclass]
+    impl ObjectSubclass for BoxedAnyObject {
+        const NAME: &'static str = "BoxedAnyObject";
+        type Type = super::BoxedAnyObject;
+    }
+    impl Default for BoxedAnyObject {
+        fn default() -> Self {
+            Self {
+                value: RefCell::new(Box::new(None::<usize>)),
+            }
+        }
+    }
+    impl ObjectImpl for BoxedAnyObject {}
+}
+
+wrapper! {
+    // rustdoc-stripper-ignore-next
+    /// This is a subclass of `glib::object::Object` capable of storing any Rust type.
+    /// It let's you insert a Rust type anywhere a `glib::object::Object` is needed.
+    /// The inserted value can then be borrowed as a Rust type, by using the various
+    /// provided methods.
+    ///
+    /// # Examples
+    /// ```
+    /// use glib::BoxedAnyObject;
+    /// struct Author {
+    ///     name: String,
+    ///     subscribers: usize
+    /// }
+    /// // BoxedAnyObject can contain any custom type
+    /// let boxed = BoxedAnyObject::new(Author {
+    ///     name: String::from("GLibAuthor"),
+    ///     subscribers: 1000
+    /// });
+    ///
+    /// // The boxed data can be stored as a `glib::object::Object`
+    /// let list = gio::ListStore::new(BoxedAnyObject::static_type());
```

Comment:
This can't compile of course as we're in `glib` and `gio` can't have been compiled so far. @GuillaumeGomez what to do here? Just `no_compile`?